### DISCUSSION TITLE
Access table without symbol / Multi Insert with array instead of hash

### DISCUSSION THREAD

**Discussion by marek22k** [jeremyevans/sequel#2055] [General]
Hello,

currently I have the following code:
```ruby
@db.create_table! "prefixorigins#{@id}" do
  column :prefix, 'cidr', primary_key: true
  column :asns, 'asn[]'
end
DB["prefixorigins#{@id}".to_sym].multi_insert @data.map { |entry| {prefix: entry[0], asns: Sequel.pg_array(entry[1])} }
```

With `create_table!` I can specify the table name as a string, but not when inserting the data. Is this how it should be? Is there a way to specify it as a string there as well?

The second question is if there is a nicer way than my `.map` to insert data into a table? Basically I have an array, however I need to convert that array to a hash so I can insert it. Is there a better way there? Maybe keeping it in an array and specifying the data by column order?
```ruby
[
    [to_insert_in_prefix, to_insert_in_asns],
    [to_insert_in_prefix, to_insert_in_asns]
]
```
vs
```ruby
[
  { prefix: to_insert_in_prefix, asns: to_insert_in_asns },
  { prefix: to_insert_in_prefix, asns: to_insert_in_asns }
]
```

**Comment by jeremyevans** [comment]
> With `create_table!` I can specify the table name as a string, but not when inserting the data. Is this how it should be? Is there a way to specify it as a string there as well?

This is expected.  In general, you should always use a symbol or Sequel-specific object for table names (`Sequel::SQL::Identifier`, `Sequel::SQL::QualifiedIdentifier`).  `create_table!` will treat a string as a symbol for backwards compatibility.  However, `Database#[]` treats a string as raw SQL code.

> The second question is if there is a nicer way than my `.map` to insert data into a table? Basically I have an array, however I need to convert that array to a hash so I can insert it. Is there a better way there? Maybe keeping it in an array and specifying the data by column order?

You probably want to use `import`:

```ruby
table = "prefixorigins#{@id}".to_sym
@db.create_table! table do
  column :prefix, 'cidr', primary_key: true
  column :asns, 'asn[]'
end
DB[table].import([:prefix, :asns], @data.map { |entry| [entry[0], Sequel.pg_array(entry[1])]})
```

**Reply by marek22k** [reply]
Thanks!
